packages/renderer/src/lib/dashboard/DashboardPage.svelte

```diff
@@ -1,18 +1,205 @@
+  if (isInitialized || isLoading) return;
+
+  isLoading = true;
+  try {
+    if (dashboardPageRegistry.entries.length > 0) {
+      const loadedItems = await loadDashboardConfiguration();
+      dashboardSections = loadedItems;
+    }
+    isInitialized = true;
+  } catch (error: unknown) {
+    console.error('Failed to load dashboard configuration:', error);
+    // Fallback to default configuration
+    dashboardSections = getDefaultDashboardItems();
+    isInitialized = true;
+  } finally {
+    isLoading = false;
+  }
+}
+
+// Load configuration from settings (like Table.svelte does)
+async function loadDashboardConfiguration(): Promise<ListOrganizerItem[]> {
+  if (tablePersistence.storage) {
+    const loadedItems = await tablePersistence.storage.load('dashboard', defaultSectionNames);
+
+    if (loadedItems.length > 0) {
+      // Ensure loaded items have proper originalOrder from defaults if missing
+      const defaultItems = getDefaultDashboardItems();
+      const items = loadedItems.map((item: ListOrganizerItem) => ({
+        ...item,
+        originalOrder: item.originalOrder ?? defaultItems.find(d => d.id === item.id)?.originalOrder ?? 0,
+      }));
+
+      // Build ordering map from loaded items
+      // Check if items are in a different order than their original order
+      const isReordered = items.some((item: ListOrganizerItem, index: number) => item.originalOrder !== index);
+      if (isReordered) {
+        const ordering = new SvelteMap<string, number>();
+        items.forEach((item: ListOrganizerItem, index: number) => {
+          ordering.set(item.id, index);
+        });
+        dashboardOrdering = ordering;
+      } else {
+        dashboardOrdering.clear();
+      }
+
+      return items;
+    }
+  }
+  return getDefaultDashboardItems();
+}
+
+// Get ordered dashboard sections based on current ordering
+function getOrderedDashboardSections(): ListOrganizerItem[] {
+  if (dashboardOrdering.size === 0) {
+    return dashboardSections;
+  }
+  return [...dashboardSections].toSorted((a, b) => {
+    const aOrder = dashboardOrdering.get(a.id) ?? a.originalOrder;
+    const bOrder = dashboardOrdering.get(b.id) ?? b.originalOrder;
+    return aOrder - bOrder;
+  });
+}
+
+// Save configuration (like Table.svelte does)
+async function saveDashboardConfiguration(): Promise<void> {
+  if (tablePersistence.storage) {
+    const orderedItems = getOrderedDashboardSections();
+    const serializableItems = orderedItems.map(item => ({
+      id: item.id,
+      label: item.label,
+      enabled: item.enabled,
+      originalOrder: item.originalOrder,
+    }));
+
+    await tablePersistence.storage.save('dashboard', serializableItems);
+  }
+}
+
+// Initialize dashboard on mount
+onMount(async () => {
+  await initializeDashboard();
+});
+
+// Save configuration whenever dashboardSections or ordering changes
+$effect(() => {
+  if (isInitialized && dashboardSections.length > 0) {
```

Comment:
issue: when adding a log above this if, it feels like we are going in this effect a lot

this could hurt performance

I will try to investigate why.